File: leetcode/38. Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string res = "1";
        for(int i = 2; i <= n; ++i){
            string tmp = "";
            int j = 0;
            while(j < res.size()){
                int cnt = 1;
                while(j != res.size() - 1 && res[j] == res[j + 1]){
                    ++cnt;
                    ++j;
                }
                tmp += to_string(cnt) + res[j];
                ++j;
            }
            if(j == res.size() - 1){
                tmp += to_string(1) + res[res.size() - 1];
            }
            res = tmp;
        }
        return res;
    }
};

File: leetcode/38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        curr = '1'

        for _ in range(1, n):
            i = 0
            j = 0
            next_curr = ''
            while i < len(curr):
                j = i+1
                while j < len(curr) and curr[j] == curr[j-1]:
                    j += 1
                next_curr += str(j-i) + curr[i]
                i = j
            curr = next_curr
        return curr
        
